pkg/api/types.go

```diff
@@ -2070,10 +2070,41 @@ const (
 	StrategicMergePatchType PatchType = "application/strategic-merge-patch+json"
 )
 
+// Type and constants for component type.
+type ComponentType string
+
+// These are the valid conditions for the component type.
+const (
+	ComponentAPIServer         ComponentType = "apiserver"
+	ComponentControllerManager ComponentType = "controller-manager"
```

Comment:
Can you expand on that?  You have a field of this type.  Is the field
holding one of these literals or a string prefixed by one of these
literals?  Is it legit to have multiple instances of these literals?

On Wed, Aug 5, 2015 at 9:38 PM, Karl Isenberg <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/12324#discussion_r36380758>
> :
>
> > @@ -2070,10 +2070,41 @@ const (
> >  	StrategicMergePatchType PatchType = "application/strategic-merge-patch+json"
> >  )
> >
> > +// Type and constants for component type.
> > +type ComponentType string
> > +
> > +// These are the valid conditions for the component type.
> > +const (
> > +	ComponentAPIServer         ComponentType = "apiserver"
> > +	ComponentControllerManager ComponentType = "controller-manager"
>
> the constant types are used as unique name prefixes, the same way pod
> names are.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/12324/files#r36380758>
> .
>